crates/sui-move-build/src/lib.rs

```diff
@@ -196,42 +186,57 @@ impl BuildConfig {
         Ok((compiled_pkg, fn_info.unwrap()))
     }
 
+    pub async fn build_async(self, path: &Path) -> anyhow::Result<CompiledPackage> {
+        let env = find_env::<SuiFlavor>(path, &self.config)?;
+        let mut root_pkg =
+            RootPackage::<SuiFlavor>::load(path.to_path_buf(), env, self.config.mode_set()).await?;
+
+        self.internal_build(&mut root_pkg)
+    }
+
     /// Given a `path` and a `build_config`, build the package in that path, including its dependencies.
     /// If we are building the Sui framework, we skip the check that the addresses should be 0
-    pub fn build(self, path: &Path) -> SuiResult<CompiledPackage> {
-        let print_diags_to_stderr = self.print_diags_to_stderr;
-        let run_bytecode_verifier = self.run_bytecode_verifier;
-        let chain_id = self.chain_id.clone();
-        let resolution_graph = self.resolution_graph(path, chain_id.clone())?;
-        build_from_resolution_graph(
-            resolution_graph,
-            run_bytecode_verifier,
-            print_diags_to_stderr,
-            chain_id,
-        )
-    }
+    pub fn build(self, path: &Path) -> anyhow::Result<CompiledPackage> {
```

Comment:
As a follow-up, it would be good to eliminate this choice for people (between `build_async` and `build`).